src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/GenericCompositeBindingProvider.cs

```diff
@@ -37,7 +37,7 @@ public GenericCompositeBindingProvider(IEnumerable<IBindingProvider> providers)
         {
             var attr = context.Parameter.GetCustomAttribute<TAttribute>();
 
-            if (attr == null)
+            if (attr == null || attr.GetType() != typeof(TAttribute))
```

Comment:
> || attr.GetType() != typeof(TAttribute)) [](start = 28, length = 41)

we should add a unit test for this too.

Reply:
True. I've been holding off on doing more work on this PR, as I am not convinced that this fix is the best way to approach inheritance in binding attributes.